python/gcp_client/hydrotools/gcp_client/gcp.py

```diff
@@ -46,7 +49,9 @@ def __init__(
         self, 
         bucket_name: str = 'national-water-model', 
         max_processes: int = None,
-        location_metadata_mapping: pd.DataFrame = None
+        location_metadata_mapping: pd.DataFrame = None,
```

Comment:
To preface this, I am also beginning to change the way I handle keyword arguments... With that being said, it may be in our best interest and the interest of any users that may start using this, to make default keyword arguments keyword only args. Im sure you are aware that in python these although are visually "keyword" arguments, python will still implicitly treat them as positional arguments if you don't provide the variable name in the call. So, to make it an explicit keyword arg you have to do the following:

```python
def function(positional_only, /, positional_or_keyword="arg1", *, keyword_only="arg2"):
  ...
```

Reply:
I think we should start adopting this way of writing and maintaining method signatures so this kind of change would not be a breaking change to the API. Thoughts?

Reply:
Some references:
https://docs.python.org/3/whatsnew/3.8.html#positional-only-parameters
https://www.python.org/dev/peps/pep-0570/

Seems like a design choice. This is a case where I could see this being the signature:
```python
def __init__(bucket_name="national_water_model", /, max_processes=None, *, location_metadata_mapping, ...):
```
But where exactly to place `/` and `*` is up for debate. The ability to fix the API with positional-only parameters is a boon, but there would still need to be thought put in to how you want the API to actually function. It's nice to have this option, but I'm not sure if it represents something we would apply broadly (at least not yet). I'd like to experiment more with these options to get a feel for the design patterns and uses that would benefit most (as well as those that might not).

Adopting this would lock us into Python 3.8+ right?

Reply:
I should have excluded the positional only argument from my example (`/`), I think my inclusion probably added confusion. I was just thinking about the keyword only arguments vs keyword/positional arguments (see [PEP 3102](https://www.python.org/dev/peps/pep-3102/)). I think adopting this pattern would lead to less breaking changes down the road. So, you above example would be:


```python
def __init__(bucket_name="national_water_model", max_processes=None, *, location_metadata_mapping, ...):
```

The positional only feature is 3.8+ but keyword only was introduced in 3.0.

Reply:
gotcha. implemented